File: firefly-example/src/main/java/com/firefly/example/reactive/coffee/store/dao/UserDAO.java
package com.firefly.example.reactive.coffee.store.dao;

import com.firefly.example.reactive.coffee.store.model.User;
import reactor.core.publisher.Mono;

/**
 * @author Pengtao Qiu
 */
public interface UserDAO {

    Mono<User> getByName(String name);

}

File: firefly-kotlin-ext/src/test/kotlin/com/firefly/kotlin/ext/example/task/management/dao/UserDao.kt
package com.firefly.kotlin.ext.example.task.management.dao

import com.firefly.kotlin.ext.example.task.management.model.User

/**
 * @author Pengtao Qiu
 */
interface UserDao {

    suspend fun insert(user: User): Long?

    suspend fun listUsers(userIdList: List<Long>): List<User>

}